disnake/poll.py

```diff
@@ -0,0 +1,415 @@
+                "This object was manually builded. To use this method you need to get a poll from the discord API!"
+            )
+
+        return PollAnswerIterator(self.poll.message, self.id, limit=limit, after=after)
+
+
+class Poll:
+    """Represents a poll from Discord.
+
+    .. versionadded:: 2.10
+
+    Parameters
+    ----------
+    question: :class:`PollMedia`
+        The question of the poll.
+    answers: List[Union[:class:`str`, :class:`PollAnswer`]]
+        The answers for this poll, up to 10.
+    duration: :class:`datetime.timedelta`
+        The total duration of the poll, up to 32 days. Defaults to 1 day.
+        Note that this gets rounded down to the closest hour.
+    allow_multiselect: :class:`bool`
+        Whether users will be able to pick more than one answer. Defaults to ``False``.
+    layout_type: :class:`PollLayoutType`
+        The layout type of the poll. Defaults to :attr:`PollLayoutType.default`.
+
+    Attributes
+    ----------
+    channel: Optional[Union[:class:`TextChannel`, :class:`VoiceChannel`, :class:`StageChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]]
+        The channel that the poll was sent from. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+
+        .. note::
+
+            This attribute is available only if you fetched the Poll object from the API.
+
+    message: Optional[:class:`Message`]
+        The message which contain this poll. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+
+        .. note::
+
+            This attribute is available only if you fetched the Poll object from the API.
+
+    question: Union[:class:`str`, :class:`PollMedia`]
+        The question of the poll.
+    duration: Optional[:class:`datetime.timedelta`]
+        The remaining duration for this poll. ``None`` if already expired.
+    allow_multiselect: :class:`bool`
+        Whether users are able to pick more than one answer.
+    layout_type: :class:`PollLayoutType`
+        The type of the layout of the poll.
+    is_finalized: :class:`bool`
+        Whether the votes have been precisely counted.
+    """
+
+    __slots__ = (
+        "_state",
+        "channel",
+        "message",
+        "question",
+        "_answers",
+        "duration",
+        "allow_multiselect",
+        "layout_type",
+        "is_finalized",
+    )
+
+    def __init__(
+        self,
+        question: Union[str, PollMedia],
+        *,
+        answers: List[Union[str, PollAnswer]],
+        duration: timedelta = timedelta(hours=24),
+        allow_multiselect: bool = False,
+        layout_type: PollLayoutType = PollLayoutType.default,
+    ) -> None:
+        self._state: Optional[ConnectionState] = None
+        self.channel: Optional[MessageableChannel] = None
+        self.message: Optional[Message] = None
```

Comment:
Both `_state` and `channel` can be obtained from `message`, do these actually need to be stored?

Reply:
Mh i think it makes more sense  to keep `_state` and remove `message` and `channel`, there's only one way to get a `Poll` direclty which is by fetching the message poll, so the user always have access to the message (and channel too)

Reply:
Joking, it's more useful to have the message object because this way we can get the original Poll duration even if the API doesn't return it